### PR TITLE
feat(go): add new CDK examples

### LINKED ISSUE #707: feat(go): adding more cdk v2 examples to go section
**🚀 Feature Request**
-----------------------
### Describe the feature

**Description**
The go section has lack of examples. I would appreciate more go examples provided here.

**Environment**
CDK CLI Version: 2.3.0 (build beaa5b2)
Language: go 1.16
OS: all

### Use Case

Since go has become popular with lambda function, it would be very helpful to provide some common cdk v2 go templates.

### Proposed Solution

**I can provide a PR with the following cdk v2 examples using go:**
- http-api &rarr; lambda-handler
- sqs &rarr; lambda-handler
- eventbridge &rarr; sqs &rarr; lambda-handler
- lambda-handler &rarr; dynamodb
- http-api &rarr;  lambda-handler protected with jwt authorizer using cognito as identity provider
- http-api &rarr;  eventbridge integration protected with jwt authorizer using cognito as identity provider
- static-website using S3, CloudFront, Certificate Manager and Route53

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### Language

Go

### PR DESCRIPTION
**Pull request by fynnfluegge**
Closes #707 

This PR contains the following cdk go excamples:
- httpapi-lambda
- httpapi-eventbridge
- cognito-httpapi
- sqs-lambda
- eventbridge-sqs
- lambda-dynamodb
- restapi-sqs
- restapi-lambda

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

**Comment by fynnfluegge** [comment]
https://github.com/orgs/aws-samples/teams/aws-cdk-maintainers Could someone pls review this PR?

**Comment by fynnfluegge** [comment]
Is anyone able to review this? Or should I split this into smaller pieces, like one PR for each example?

**Comment by fynnfluegge** [comment]
Hey @peterwoodworth, is it possible that someone will review this? I removed the ```cloudfront-s3-website``` example since there occured some issues with the jsii docker build recently. It requires an aws account id to fetch a hosted zone. However, the remaining eight examples are working properly. Looking forward to hear from you 🙌🙂

**Comment by peterwoodworth** [comment]
Hey @fynnfluegge, thanks again a ton for this contribution. We need some Go examples!

I'm not great at Go however, so I don't feel comfortable reviewing this one. I'm going to suggest that @indrora review this 🙂 If they aren't able to review this soon, then @indrora could you request that @pahud take a look? Thanks

File: go/cognito-httpapi/README.md
Comment (indrora) [inline go/cognito-httpapi/README.md:1]: Please expand this. 

There's very little content here. We generally like larger, end to end covering from "want to do" through to "end up like this".

Tell me what this example does. Tell me how to deploy it. Tell me how I know it worked.

File: go/httpapi-lambda/README.md
Comment (indrora) [inline go/httpapi-lambda/README.md:1]: please expand this readme to include information from other readmes, at least. 

Go is no longer in developer preview.

File: go/restapi-sqs/README.md
Comment (indrora) [inline go/restapi-sqs/README.md:1]: There's more to say here. 

Explain yourself.

**Review by indrora — CHANGES_REQUESTED** [review]
Needs better READMEs overall. please expand your readmes to include, at minimum,

* What this example aims to demonstrate
* What the reader needs to do in order to deploy this example
* What the reader does to verify that the example is working.

Code comments are sparse. Add comments to each example's main at least explaining what the code does. In cases where you've had to extend a construct or type, document those as appropriate.

**Comment by fynnfluegge** [comment]
@indrora Many thanks for your feedback, will provide more comments and extends the READMEs asap!

**Comment by fynnfluegge** [comment]
Hey @indrora, I added code comments and extended all READMEs 👌

File: go/httpapi-eventbridge/README.md
Comment (indrora) [inline go/httpapi-eventbridge/README.md:11]: Explain here how to get the endpoint of the stack.

File: go/lambda-dynamodb/README.md
Comment (indrora) [inline go/lambda-dynamodb/README.md:27]: placeholder text doesn't match where it's used, getting the function ARN should be described here as well.

**Review by indrora — DISMISSED** [review]

**Comment by indrora** [comment]
Looks great. 

My only big nitpick is that go 1.19 is the earliest supported version of Go currently, and most of this targets go 1.16. This is 100% the fault of this PR just being Old:tm: and not really any person's fault. 

This can be addressed in a later change, and should be done on a regular basis as a chore.

**Comment by fynnfluegge** [comment]
> Looks great.
> 
> My only big nitpick is that go 1.19 is the earliest supported version of Go currently, and most of this targets go 1.16. This is 100% the fault of this PR just being Old™️ and not really any person's fault.
> 
> This can be addressed in a later change, and should be done on a regular basis as a chore.

@indrora Solved your README suggestions ✅ You're right 1.19 would be more appropriate, but the PR is indeed few month Old™, I will create a new PR once this is merged and update all go versions to the latest 👌

File: go/lambda-dynamodb/README.md
Comment (fynnfluegge) [reply go/lambda-dynamodb/README.md]: True, I forgot to log the function ARN during `cdk delpoy`. Good hint 👍

File: go/httpapi-eventbridge/README.md
Comment (fynnfluegge) [reply go/httpapi-eventbridge/README.md:11]: Added a hint that the endpoint is logged during `cdk deploy` ✅

**Review by indrora — APPROVED** [review]

**Review by indrora — APPROVED** [review]

**Pull request merged**
